apps/ima/behaviours/save-form-session.js

```diff
@@ -75,15 +75,33 @@ module.exports = superclass => class extends superclass {
 
         const isContinueOnEdit = req.form.options.continueOnEdit &&
           _.get(req.form.options.forks, '[0].continueOnEdit');
-        // Redirect to final-summary if on the edit flow of the harm-claim page
-        if((req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1] === '/harm-claim-summary'
-        || req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1] === '/harm-claim' ||
-        req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1] === '/harm-claim-countries' ||
-        req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1].includes('/risk-of-harm') ||
-        req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1].includes('/harm-claim-details')) &&
-        (req.form.options.continueOnEdit === false || req.sessionModel.get('is-serious-and-irreversible') === 'no') &&
-        req.sessionModel.get('steps').includes('/human-rights-claim')) {
-          return res.redirect('/ima/final-summary');
+
+        const sessionSteps = req.sessionModel.get('steps');
+        const formSteps = Object.keys(req.form.options.steps);
+        const orderedSessionSteps = formSteps.filter(step => sessionSteps.includes(step));
+        const latestStepInJourney = orderedSessionSteps[orderedSessionSteps.length - 1];
+
+        // logic to override the SIH routing to resolve issues with routing
+        if(JSON.stringify(req.form.options.route, null, 2) === '"/harm-claim"' &&
+        req.sessionModel.get('is-serious-and-irreversible') === 'yes' &&
+        req.sessionModel.get('sih-countries') !== undefined) {
+          if(req.sessionModel.get('sih-countries').aggregatedValues.length === 5) {
+            return res.redirect('/ima/harm-claim-summary');
+          }
+          return res.redirect('/ima/harm-claim-countries');
+        } else if(JSON.stringify(req.form.options.route, null, 2).includes('/harm-claim-summary')) {
+          if(req.sessionModel.get('steps').includes('/evidence-upload')) {
+            return res.redirect('/ima/final-summary');
+          } else if(!req.sessionModel.get('steps').includes('/human-rights-claim')) {
+            return res.redirect('/ima/human-rights-claim');
+          }
+          return res.redirect('/ima/' + latestStepInJourney);
+        } else if (JSON.stringify(req.form.options.route, null, 2) === ('"/harm-claim"') &&
```

Comment:
Does it need to be stringified? Could it not just be ` if(req.form.options.route) === '/harm-claim'`?

Reply:
Same as comment above.